### PR TITLE
My Code Review Suggestions

### PR DESCRIPTION
**Pull request by Emiller321**

### CODE REVIEW

**Comment by Emiller321** [comment]
**Analysis of the program.**
Answer: This program allows the user to tell the time the sun will set and rise each day from any location and tells the weather. Very cool app for people that need that information known at a given time

**Was the program available in UC Github on time?**
Answer: Yes the program was readily available to me and on time

**Is the program documented/commented well enough for you to understand?**
Answer: Yes code was mainly clean and easy to understand. Files are very organized.

**Does the program compile?**
Answer: There was a failing check in circle ci build

**Rationale behind your changes.**
Answer: The three changes that I have made was first changing a comment to a proper multi line comment instead of a bunch of single line comments all together. Second was implementing the use of a trailing comma because of the benefits that it provides where it makes the version control diffs cleaner and it makes it easy to add and reorder elements. Although these are entirely optional it does have its benefits. Last, was fixing spacing when it comes with commas where there was a space before a colon where it separates a declaration and its type. There should be no spacing before a colon when it separates a declaration and its type.

File: app/src/main/java/com/sundial/v1001/LocationLiveData.kt [hunk 1]
```diff
@@ -27,13 +27,13 @@ class LocationLiveData(var context: Context) : LiveData<LocationDetails>() {
...
             ) != PackageManager.PERMISSION_GRANTED
         ) {
-            // TODO: Consider calling
-            //    ActivityCompat#requestPermissions
-            // here to request the missing permissions, and then overriding
-            //   public void onRequestPermissionsResult(int requestCode, String[] permissions,
-            //                                          int[] grantResults)
-            // to handle the case where the user grants the permission. See the documentation
-            // for ActivityCompat#requestPermissions for more details.
+            /* TODO: Consider calling
```
Comment (Emiller321) [inline app/src/main/java/com/sundial/v1001/LocationLiveData.kt:30]: Implemented Multi line comment instead of a bunch of single line comments 

Link: https://www.tutorialspoint.com/kotlin/kotlin_comments.htm#:~:text=Kotlin%20Multi%2Dline%20Comments,be%20ignored%20by%20Kotlin%20compiler.

File: app/src/main/java/com/sundial/v1001/dao/ITwilightDAO.kt [hunk 2]
```diff
@@ -5,7 +5,6 @@ import retrofit2.Call
 import retrofit2.http.GET
 
 interface ITwilightDAO {
-
```
Comment (Emiller321) [inline app/src/main/java/com/sundial/v1001/dao/ITwilightDAO.kt:8]: Eliminated unnecessary whitespace

File: app/src/main/java/com/sundial/v1001/dto/LocationDetails.kt [hunk 3]
```diff
@@ -1,5 +1,3 @@
 package com.sundial.v1001.dto
 
-class LocationDetails(val longitude : String, val latitude : String) {
-
-}
+class LocationDetails(val longitude: String, val latitude: String,) {}
```
Comment (Emiller321) [inline app/src/main/java/com/sundial/v1001/dto/LocationDetails.kt:3]: Eliminated unnecessary whitespace

Comment (Emiller321) [inline app/src/main/java/com/sundial/v1001/dto/LocationDetails.kt:3]: Implemented a trailing comma benfits can be found in link provided
https://kotlinlang.org/docs/coding-conventions.html#trailing-commas

File: app/src/main/java/com/sundial/v1001/dto/Twilight.kt [hunk 4]
```diff
@@ -3,6 +3,4 @@ package com.sundial.v1001.dto
 import java.util.Date
 import com.google.gson.annotations.SerializedName
 
-data class Twilight (@SerializedName("sunrise") var sunrise : Date, @SerializedName("sunset") var sunset : Date ){
-
-}
\ No newline at end of file
+data class Twilight (@SerializedName("sunrise") var sunrise: Date, @SerializedName("sunset") var sunset: Date, ){}
```
Comment (Emiller321) [inline app/src/main/java/com/sundial/v1001/dto/Twilight.kt:6]: Implemented a trailing comma, and fixed colon spacing 
https://kotlinlang.org/docs/coding-conventions.html#colon

File: app/src/main/java/com/sundial/v1001/dto/User.kt [hunk 5]
```diff
@@ -1,5 +1,3 @@
 package com.sundial.v1001.dto
 
-data class User(val uid: String = "", var displayName: String?){
-
-}
\ No newline at end of file
+data class User(val uid: String = "", var displayName: String?){}
```
Comment (Emiller321) [inline app/src/main/java/com/sundial/v1001/dto/User.kt:3]: Fixed colon spacing per suggestions in link
https://kotlinlang.org/docs/coding-conventions.html#colon

File: app/src/main/java/com/sundial/v1001/LocationLiveData.kt (hunk 1 above)
Comment (discospiff) [reply app/src/main/java/com/sundial/v1001/LocationLiveData.kt:30]: Actually...

KDoc format belongs only above public classes and functions.  But I would do niether.

In this case, this is a discussion.  A better approach is to move this conversation to a story on the sprint board, and/or create a branch for it.  Put the conversation where developers can have a dialog on it, not in the source code, where it is clutter.

File: app/src/main/java/com/sundial/v1001/dao/ITwilightDAO.kt (hunk 2 above)
Comment (discospiff) [reply app/src/main/java/com/sundial/v1001/dao/ITwilightDAO.kt:8]: Formatting can, and should, be performed automatically, and consistently, by the IDE.  Formatting code is really programmer's preference anyway, and doesn't necessarily decrease technical debt.

File: app/src/main/java/com/sundial/v1001/dto/LocationDetails.kt (hunk 3 above)
Comment (discospiff) [reply app/src/main/java/com/sundial/v1001/dto/LocationDetails.kt:3]: Formatting can, and should, be performed automatically, and consistently, by the IDE.  Formatting code is really programmer's preference anyway, and doesn't necessarily decrease technical debt.

File: app/src/main/java/com/sundial/v1001/dto/User.kt (hunk 5 above)
Comment (discospiff) [reply app/src/main/java/com/sundial/v1001/dto/User.kt:3]: Formatting can, and should, be performed automatically, and consistently, by the IDE.  Formatting code is really programmer's preference anyway, and doesn't necessarily decrease technical debt.

**Comment by discospiff** [comment]
This branch is mostly formatting, with no significant tech debt reduction.  No need to merge.
